### ISSUE TITLE
Undocumented PDF, quantile, CDF, and random deviate functions

### ISSUE THREAD

**Issue by davidcerny** [revbayes/revbayes#526]
Our `basics.Rev` test script (see `tests/test_basics/scripts`) shows that at least for several basic math distributions (uniform, exponential, normal, lognormal, gamma, beta, and – where applicable – Dirichlet), we have R-like functions for obtaining the probability density (prefixed with `d`, e.g., `dunif`, `dexp`, etc.), quantile (prefixed with `q`), cumulative distribution function (prefixed with `p`), and random deviates (prefixed with `r`). These often represent neat alternatives to the usual Rev syntax: for example,

```
a ~ dnExp( 1.0 )
a.clamp( 0.7789 )
a.lnProbability()
```

can be more concisely written as

```
dexp( 0.7789, 1.0 )
```

and the following

```
for (i in 1:5) {
    a[i] ~ dnExp( 1.0 )
}
a
```

is equivalent to:

```
rexp(5, 1.0)
```

Note, however, that these random deviates are treated as constant rather than stochastic variables.

There are several documentation-related problems I can see here:

1. There is no way to even learn of the existence of these functions; they are not included in our [online documentation](https://revbayes.github.io/documentation/), and I'm still unsure which distributions have them. `dnBirthDeath` does, for example, since `rBirthDeath` or `rBDP` are valid functions for simulating birth-death trees. But `dnInverseGamma` doesn't seem to, since `rInverseGamma` does not exist, and neither does `rinversegamma` or `rinvgamma`.

2. If you type `?rexp` into the Rev console, you get the usual help page with aliases, usage, and arguments, but there is no actual explanation of what the function is doing. That can only be inferred from the test script mentioned above, and from the fact it shares its name and behavior with the equivalent R function.

3. Speaking of which, it's a mystery to me where these help pages are actually pulled from. I searched through my whole local RevBayes directory for occurrences of the string "rexponential" (`find . -type f -exec grep -l "rexponential" {} \;`), and the search came up empty – what am I missing here?

**Comment by bjoelle** [comment]
to answer 3. - it's set as a template based on the distribution name I believe, see e.g. `DistributionFunctionCdf::getFunctionNameAliases` or `DistributionFunctionRv<valueType>::getFunctionNameAliases`

**Comment by brpetrucci** [comment]
@davidcerny I assume this issue should still stay open to remind us to make help files for these functions eventually, but are your questions about help pages answered? We've talked a bit about it a few months ago and I want to make sure it all makes sense now. On that note, maybe we should document how our documentation is built somewhere

**Comment by bredelings** [comment]
Inspecting the existing functions using `ls(TRUE)` and looking under "Uncategorized", it looks like maybe the `rX` form is being autogenerated for all distributions?  For inverse gamma, the name is `rinverseGamma`.  Possibly the code that autogenerates `dnInverseGamma` upcases the first letter, but the code that generates `rinverseGamma` does not?

**Comment by bredelings** [comment]
I guess it makes more  sense now that it seems that they are consistently available.

**Comment by jembrown** [comment]
I use these functions (especially pdf and random deviate) all the time for teaching.

**Comment by brpetrucci** [comment]
I agree they would ideally be methods on distribution objects. Maybe that can be a future goal, and documenting the existing functions can be a more achievable short-term one
